### PR TITLE
Remove generic advice about comments

### PR DESCRIPTION
**Pull request by andyw8**
These are common to good coding practices, not specific to Elixir.

I think it's better to remove them to keep the guide focused and concise.

### CODE REVIEW

**Comment by asaaki** [comment]
But does it any harm if we keep them?
There will be always new developers not knowing all the commons yet.

File: README.md [hunk 1]
```diff
@@ -446,13 +446,6 @@ def some_function(_),
   String.upcase(some_string) # Capitalize string.
   ```
 
-* Keep existing comments up-to-date.
-  An outdated comment is worse than no comment at all.
-
-* Avoid writing comments to explain bad code.
-  Refactor the code to make it self-explanatory.
-  ("Do — or do not — there is no try." —Yoda)
```
Comment (eksperimental) [inline README.md]: I dont even get what ths comment is trying to convey

**Comment by ericmj** [comment]
> But does it any harm if we keep them?
There will be always new developers not knowing all the commons yet.

Because it keeps the guide unfocused. If we include this why not include every general programming advice useful for beginners?

**Comment by andyw8** [comment]
(Hello Christoph!)

True, it doesn't do much harm, but it does perhaps set a precedent about what's suitable for inclusion.

One option could could be to link to some external advice about good coding, rather than duplicating that information in the guide?

**Comment by asaaki** [comment]
(Hello Andy!)

I still believe it is related and noteworthy enough, especially in context of the whole "Comments" section.
Otherwise you could argue even that the whole section could be deleted as well, since

> Write self-documenting code and ignore the rest of this section. Seriously!

wraps it up pretty nicely. Just don't write comments. 😉

**Comment by ericmj** [comment]
I agree @asaaki, :+1: to remove the whole section. Most of the guidelines seems to be not specific to Elixir or very specific.

**Comment by whatyouhide** [comment]
(also, "don't write comments" is a bit of a silly thing to say because code comments are generally good, they can provide context to newcomers to some project, can justify apparently weird choices, and so on 😛)

**Comment by BenBals** [comment]
As far as I am concerned the comment philosophy is a very important aspect of coding style and as such has it's right to be in a style guide. There are many project which would probably would have turned out better if instead of writing the explanatory comment the code would have been refactored to be expressive itself.

**Comment by ericmj** [comment]
There's lots of general programming advice that can improve the quality and code base of projects. Many, many books have been written of the subject.

The question is what the scope of "The Elixir Style Guide" is. Should it be specifically about the style of elixir code or should it also include general programming advice, a subject which is essentially limitless and has lots of differing opinions?

**Comment by BenBals** [comment]
@ericmj You are right on that. This might not be the right place for advice like this. It would expose many starters to advice I'd width I'd had been given when starting out. This information is available, but beginners don't just read "Clean Code" or the like. So I think we should find a place where we, as the Elixir community, could give such advise in an accessible and easy to understand way such that beginners are exposed to it early.

**Comment by tobyhinloopen** [comment]
In what world are we living where beginners are reading style guides? The comment advice is of little use for beginners since there is no explanation why comments should be avoided.

I would still leave it though, just as a friendly reminder for those who care

**Comment by christopheradams** [comment]
I count 7 comments or reactions in support of this PR, and 3 either against or ambivalent. The ayes have it.

If anyone feels the whole section should be removed or reworked, please open a new PR.

**Pull request merged**